zeppelin-plugins/notebookrepo/mongo/src/main/java/org/apache/zeppelin/notebook/repo/MongoPlugin.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo;
+
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MongoPlugin extends Plugin {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoPlugin.class);
+
+  public MongoPlugin(PluginWrapper wrapper) {
+    super(wrapper);
+  }
+
+  @Override
+  public void start() {
+    LOGGER.info("MongoPlugin started");
```

Comment:
It looks like all the Plugin start and stop method are the same, then how about defining a Base Plugin class that implement the start and stop method. And we can just add one field `name` to this base plugin class and each plugin implementation can set value to `name` in the constructor.

Reply:
I added a new BasePlugin class that contains the general code.